client/notifications/index.jsx

```diff
@@ -155,7 +156,16 @@ export class Notifications extends Component {
 		const customMiddleware = {
 			APP_RENDER_NOTES: [ ( store, { newNoteCount } ) => this.props.setIndicator( newNoteCount ) ],
 			OPEN_LINK: [ ( store, { href } ) => window.open( href, '_blank' ) ],
-			OPEN_POST: [ ( store, { href } ) => window.open( href, '_blank' ) ],
+			OPEN_POST: [ ( store, { siteId, postId, href } ) => {
+				if ( config.isEnabled( 'notifications/link-to-reader' ) ) {
+					this.props.checkToggle();
+					const commentIdIndex = href.indexOf( '#comment-' );
+					const commentHash = commentIdIndex === -1 ? '' : href.substring( commentIdIndex );
+					page( `/read/blogs/${ siteId }/posts/${ postId }${ commentHash }` );
```

Comment:
just a note that we need to add the `OPEN_COMMENT` action instead of hacking the URL here…